### PR TITLE
Require preservation of supplied notice lines (close #36)

### LINKED ISSUE #36: Require preservation of licensor identification and contact information
I suggest we include a name or contact information for the licensor in the license.  Because limited licenses implicitly invite negotiation of alternative terms, licensees should know where to seek them.

### PR DESCRIPTION
**Pull request by kemitchell**
This PR adds text to the notice-preservation requirement of Distribution License requiring preservation of notice license identified by a specific string prefix.  I have a few things in mind:

1. Avoid turning the license text into a form with blanks, like MIT and BSD, that licensors have to "fill out". Remain a fixed license text, like GPL, Apache-2.0, or the Creative Commons licenses.

2. Give licensors the flexibility to include whatever information they like, such as identification, websites, contact methods, and notices that other licenses are available, if any.

3.  Simplify compliance by providing a single, machine-identifiable marker to test whether there are other notices to preserve, and identify them for preservation. (e.g. `grep -r -F "Required Notice:"`)

Which "magic string" to choose for the notice marker could turn into a bikeshed real quick.  I'm not wedded to `Required Notice:`, but I think anything we choose should be unambiguous, short, and generic enough for other licenses to use if they choose. I would point out, however, that as yet I have _not_ thought of this as a means of assuring the licensor "credit". That would require additional terms to make sure end users, including service users, receive the notices, and that they be at least discoverable, if not prominent. If we do proper "credit" terms, we should probably do them in 2.0.0, later.

### Apache-2.0

> 4\. Redistribution. You may reproduce and distribute copies of the Work or Derivative Works thereof in any medium, with or without modifications, and in Source or Object form, provided that You meet the following conditions:
> ...
> c. You must retain, in the Source form of any Derivative Works that You distribute, all copyright, patent, trademark, and attribution notices from the Source form of the Work, excluding those notices that do not pertain to any part of the Derivative Works; and
> d. If the Work includes a "NOTICE" text file as part of its distribution, then any Derivative Works that You distribute must include a readable copy of the attribution notices contained within such NOTICE file, excluding those notices that do not pertain to any part of the Derivative Works, in at least one of the following places: within a NOTICE text file distributed as part of the Derivative Works; within the Source form or documentation, if provided along with the Derivative Works; or, within a display generated by the Derivative Works, if and wherever such third-party notices normally appear. The contents of the NOTICE file are for informational purposes only and do not modify the License. You may add Your own attribution notices within Derivative Works that You distribute, alongside or as an addendum to the NOTICE text from the Work, provided that such additional attribution notices cannot be construed as modifying the License.

### CODE REVIEW

**Comment by kemitchell** [comment]
A few thoughts on Creative Commons' approach:

> 2. You may satisfy the conditions in Section 3(a)(1) in any reasonable manner based on the medium, means, and context in which You Share the Licensed Material. For example, it may be reasonable to satisfy the conditions by providing a URI or hyperlink to a resource that includes the required information.

I believe we adequately address this for the terms of the license itself, by allowing a hyperlink to the text on polyformproject.org. I'm not sure this clarification is necessary for notice lines. Creative Commons _did_ intend attribution of this kind more in the nature of credit, and didn't have the software application/service distinction to deal with.

> 3. If requested by the Licensor, You must remove any of the information required by Section 3(a)(1)(A) to the extent reasonably practicable.

I don't have a good citation, but I read this primarily as a concern about integrity where the license permits derivative works.  In essence, I see it as analogous to the kinds of reputation-protection conditions, for maintainers and "official" versions, that we see in Artistic-2.0 and the like.  Those protections have gone out of fashion in open software licensing, and unless we hear strongly to the contrary, I'm happy to publish Polyform licenses without them.

**Comment by kemitchell** [comment]
Related: #36

**Comment by heathermeeker** [comment]
I like this idea, but I want to make sure what it requires can easily be done automatically.  I'm sure that's what Apache was trying to do, but in practice it seems like a failure.  Can one put something in a license.txt file that automatically pulls in the additional notice information? (I'm guessing .txt cannot do that.) Apologies as I don't really know what I'm asking for, I just want to be sure we are treading on the compliance process as little as possible.

**Comment by kemitchell** [comment]
@heathermeeker I fault Apache's `NOTICE` rule not for being hard to follow, but because so few developers and compliance people know it exists.

At least as far as I'm involved in compliance, doing a plain-text search for lines with a specific string and adding them to a notice file seems pretty easy to do. Easier than Creative Commons attribution, easier than Apache 2.0 "all copyright, patent, trademark, and attribution notices from the Source form of the Work", but harder than Apache 2.0 `NOTICE` files. But that's in my little corner of the software world.

@lindenksv, what do you think of a license that requires preserving notices that appear on lines that start with a specific string, like `Required Notice:`, along with the license terms? Easy enough to administer?

**Comment by lindenksv** [comment]
I think a a specific string is good enough for any compliance tool. I would maybe call it "Required Licensing Information" rather than just "notice," though, because notices can often be unrelated to licensing and "notice" to an engineer doesn't sound like legal language they need to care about.

And I think it's best if it's super clear exactly what text needs to be preserved so people can easily copy and paste and don't have to guess. While people can add additional info, like their website or info on getting other licenses, I don't think it's practical or useful to require that that sort of info be preserved. If it happens to be after the "Required Licensing Information" tag and people copy and paste it, great. But if they leave that part out, that should be ok. I'd like to see a model notice for this license somewhere, preferably in the license. Short and sweet: "Required Licensing Information: TuskMaker is licensed under the Polyform Project License-DL, available at polyformproject.org/polyform-dl. Copyright (c) 2019 Elephant LLC. "

One question is, do you think this "Required Licensing Information" needs to be in every file or just one, NOTICE? My own feeling is that the advice should be to only mark once in the NOTICE and not in every file. Because practically speaking, the vast majority of OSS is used unmodified and marking every file actually complicates the compliance process by increasing the number of things that have to be looked at and increasing the likelihood that not all files will be marked exactly the same way, even though they mean the same thing. And, because most people, even giant corporations, don't copyright their OSS, making the notice fairly irrelevant. 

One other thing- in the polyform license you say people can link to polyformproject.org, but presumably you'll have a spate of licenses and linking to the top level domain doesn't actually tell you what license is applicable. I guarantee that as written, people will link to the top level domain and not the actual applicable license. This needs to be clearer - preferably with a model notice as I gave above. It drives me crazy that every time I want to do a CC attribution, I gotta dig around on the Wikipedia FAQ to see a model I can copy and paste.

A little rant on Apache:

Much of the time you see people labeling files (or entire projects) like this: "bobcattool is under MIT." And if they bother to do a NOTICE file or a similar attribution, they will mention the licensing in the same way. Even Apache, when they attribute other dependencies, do it like that, because they have this magical notion that there will be one set of attribution docs for the binary and another for the source, when I think in practice, there is no separate set of docs for the binary and to the extent there is, most lawyers and compliance people don't have the tools or knowledge to access them. Many compliance tools can't pull that stuff out of a binary they encounter. So, in their view, if you want to see the copyright notices, you can go dig in the source code yourself. And when they do an attribution for the binary, then someone will maybe put that in an attribution file? I dunno, call me when that actually happens.... Oh, and, even crazier, they say that that the attribution docs you find on Github don't need to attribute dependencies that come in at build time. They'll attribute those in the attribution notes for the binary....So in practice, we actually have no clue about many dependencies of Apache projects because of this policy.

**Comment by kemitchell** [comment]
A few takeaways from @lindensvk's wonderful comment:

- [ ] Push a commit changing the "magic string" to `Required Licesning Information:`.

- [ ] Add an example line using that string.

- [ ] Open a separate pull request to address the license URL.  I'll probably add a URL right after the title of the license, and reference that URL from Distribution License below.

**Comment by kemitchell** [comment]
@lindenksv: one notice versus a notice per file, I don't think I'd read the currently proposed language to require providing notices in any particular way, or multiple times. As long as recipients also get the language required, in some way not obviously calculated to make it hard to find, I'd say the licensee has met the condition.

**Comment by kemitchell** [comment]
@lindenksv 735a22acb381ca5d7e9bfc793ac765f346d2791d changes the magic string to `Required Licensing Information:`, as you suggested.

**Comment by kemitchell** [comment]
936b6cdfa7b93a868365bc820674e4f30e76e5be adds an example notice line.

My only hesitation at this point is that our magic string, `Required Licensing Information:` is pretty long.

**Comment by heathermeeker** [comment]
My experience is that almost no one will put notices in every file, whether required to or not, but we have not required it anyway.  It’s best not to get too specific, because, believe it or not, GITHUB made it not always be the primary way to distribute code. A lot of the existing open source licenses have lived through several generations of media distribution and source code repositories, and the more specific their direction, the more likely they are to become obsolete and difficult to comply with. We will eventually post some guidance on use of the licenses that is external to them— tips and tricks—and this would be a good topic to cover because the exact placement of notices is a question clients ask a lot. Not to conflate two topics, but the other subject we need to provide some guidance on is whether the doctors want to use a click to accept mechanism.

**Comment by kemitchell** [comment]
Maybe `Required Credit:`?

**Comment by lindenksv** [comment]
I don't think you need any compliance hints in the license other than to say "reproduce what comes after "required notice""

**Comment by kemitchell** [comment]
@lindenksv Thanks! I've reverted the changes adding the example command. Merging the rest.
